File: Graph_Representation/javascript/Adjacency_matrix.js
// JavaScript Implementation
// Adjacency Matrix

class Graph {
    constructor(size = 1) {
        this.size = size;
        this.matrix = [];
        for (let i = 0; i < size; i++) {
            this.matrix.push([]);
            for (let j = 0; j < size; j++) {
                this.matrix[i][j] = 0;
            }
        }
    }

    // Adding Edges
    addEdge(vertex1, vertex2, weight = 1) {
        if (vertex1 > this.size - 1 || vertex2 > this.size - 1) {
            console.log('invalid vertex');
        } else if (vertex1 === vertex2) {
            console.log('same vertex');
        } else {
            this.matrix[vertex1][vertex2] = weight;
            this.matrix[vertex2][vertex1] = weight;
        }
    }

    // Adding Nodes
    addVertex() {
        this.size++;
        this.matrix.push([]);
        for (let i = 0; i < this.size; i++) {
            this.matrix[i][this.size - 1] = 0;
            this.matrix[this.size - 1][i] = 0;
        }
    }

    // Printing adjacency list
    printMatrix(val) {
        for (let i = 1; i <= val; i++) {
            let row = '';
            for (let j = 1; j <= val; j++) {
                row += ` ${this.matrix[i][j]}`;
            }
            console.log(row);
        }
    }
}


var g = new Graph(6);
var vertices = [ '1', '2', '3', '4', '5', '6' ];

for (var i = 0; i < vertices.length; i++) {
    g.addVertex(vertices[i]);
}

g.addEdge('1', '3');
g.addEdge('1', '4');
g.addEdge('2', '4');
g.addEdge('3', '2');
g.addEdge('4', '5');
g.addEdge('5', '6');

g.printMatrix(6);

File: ACADEMIC CODES/Graph_Representation/java/Adjacency_matrix.java
// Java Implementation
// Adjacency Matrix

public class Adjacency_matrix {
  private boolean adjMatrix[][];
  private int numVertices;

  // Initialization
  public Adjacency_matrix(int numVertices) {
    this.numVertices = numVertices;
    adjMatrix = new boolean[numVertices][numVertices];
  }

  // Adding Edges
  public void addEdge(int i, int j) {
    adjMatrix[i][j] = true;
    adjMatrix[j][i] = true;
  }


  // Printing the matrix
  public String toString() {
    StringBuilder s = new StringBuilder();
    for (int i = 0; i < numVertices; i++) {
      for (boolean j : adjMatrix[i]) {
        s.append((j ? 1 : 0) + " ");
      }
      s.append("\n");
    }
    return s.toString();
  }

  public static void main(String args[]) {
    Adjacency_matrix g = new Adjacency_matrix(6);

    System.out.print("\n");
    g.addEdge(0, 2);
    g.addEdge(0, 3);

    g.addEdge(1, 3);
    g.addEdge(2, 1);

    g.addEdge(3, 4);
    g.addEdge(4, 5);

    System.out.print(g.toString());
    System.out.print("\n");
  }
}